### PR TITLE
Send email to applicant on signing

### PR DESCRIPTION
**Pull request by vrajmohan**

### CODE REVIEW

**Comment by vrajmohan** [comment]
I will update the deploy pipeline to update Mailgun config vars in a separate commit.

File: src/main/java/org/mdbenefits/app/submission/actions/HandleApplicationSigned.java [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
+    }
+
     @Override
     public void run(Submission submission) {
+        sendEmailToApplicant(submission);
+    }
+
+    private void sendEmailToApplicant(Submission submission) {
+        String recipientEmail = (String) submission.getInputData().get("emailAddress");
+        if (recipientEmail == null || recipientEmail.isBlank()) {
```
Comment (analoo) [inline src/main/java/org/mdbenefits/app/submission/actions/HandleApplicationSigned.java:36]: @vrajmohan - How would you feel about adding a check here to see what environment we are in. 

If, in staging or dev, maybe we can skip email sends, or override to send to a test mailing list instead?

We are testing a lot and I'm concerned that we might accidentally send emails out every once in a while.

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -38,6 +38,11 @@ form-flow:
       smarty_auth_id: ${SMARTY_AUTH_ID}
       smarty_auth_token: ${SMARTY_AUTH_TOKEN}
       smarty_url: "https://us-street.api.smartystreets.com/street-address"
+  email-client:
```
Comment (analoo) [inline src/main/resources/application.yaml:41]: @vrajmohan - does this configuration only apply to the production app? If so, please disregard this comment: https://github.com/codeforamerica/mdpilot/pull/40/files#r1483722700

File: src/main/java/org/mdbenefits/app/submission/actions/HandleApplicationSigned.java (hunk 1 above)
Comment (vrajmohan) [reply src/main/java/org/mdbenefits/app/submission/actions/HandleApplicationSigned.java:36]: Good point, see comment below.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (vrajmohan) [reply src/main/resources/application.yaml:41]: The configuration applies to all environments but relies on the `MAILGUN_KEY` env var being set. The email will not be sent if `MAILGUN_KEY` is incorrect or not set.

**Review by analoo — APPROVED** [review]

**Pull request merged**